code/modules/chemistry/tools/food_and_drink.dm

```diff
@@ -150,6 +150,12 @@ ABSTRACT_TYPE(/obj/item/reagent_containers/food/snacks)
 				return
 
 			src.Eat(user,user)
+		else if (istype(W, /obj/item/tongs))
+			if (istype(src.loc, /obj/item/storage))
+				boutput(user, "You take [src] out of [src.loc].")
+				user.put_in_hand_or_drop(src)
+			else
+				src.AttackSelf(user)
```

Comment:
should move the relevant code in `attack_self` to some other proc and then call that proc in `attack_self` and here; current implementation is likely fine but might cause weird issues down the line when something expects us to be using an item on itself in the usual way and we... aren't

note: a lot of existing code follows this pattern (mousedrop calling attackby, etc) which isn't ideal!